pkg/controller/deployable/deployable_controller.go

```diff
@@ -52,14 +54,127 @@ func Add(mgr manager.Manager) error {
+	authClient := kubernetes.NewForConfigOrDie(mgr.GetConfig())
 
 	return &ReconcileDeployable{
 		Client:        mgr.GetClient(),
 		scheme:        mgr.GetScheme(),
+		authClient:    authClient,
 		eventRecorder: erecorder,
 	}
 }
 
+type placementruleMapper struct {
+	client.Client
+}
+
+func (mapper *placementruleMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if klog.V(utils.QuiteLogLel) {
+		fnName := utils.GetFnName()
+		klog.Infof("Entering: %v()", fnName)
+		defer klog.Infof("Exiting: %v()", fnName)
+	}
+
+	cname := obj.Meta.GetName()
+	klog.V(10).Info("In placement Mapper:", cname)
+
+	var requests []reconcile.Request
+
+	dplList := &appv1alpha1.DeployableList{}
+	listopts := &client.ListOptions{}
+	err := mapper.List(context.TODO(), listopts, dplList)
+	if err != nil {
+		klog.Error("Failed to list deployables for placementrule mapper with error:", err)
+		return requests
+	}
+
+	for _, dpl := range dplList.Items {
+		if dpl.Spec.Placement == nil || dpl.Spec.Placement.PlacementRef == nil || dpl.Spec.Placement.PlacementRef.Name != obj.Meta.GetName() {
+			continue
+		}
+
+		annotations := dpl.GetAnnotations()
+		if annotations == nil || annotations[appv1alpha1.AnnotationManagedCluster] == "" {
+			continue
+		}
+
+		if annotations[appv1alpha1.AnnotationManagedCluster] != (types.NamespacedName{}).String() {
+			continue
+		}
+
+		// only reconcile it's own object
+		objkey := types.NamespacedName{
+			Name:      dpl.GetName(),
+			Namespace: dpl.GetNamespace(),
+		}
+		requests = append(requests, reconcile.Request{NamespacedName: objkey})
+	}
+
+	return requests
+}
+
+type clusterMapper struct {
+	client.Client
+}
+
+func (mapper *clusterMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if klog.V(utils.QuiteLogLel) {
+		fnName := utils.GetFnName()
+		klog.Infof("Entering: %v()", fnName)
+		defer klog.Infof("Exiting: %v()", fnName)
+	}
+
+	cname := obj.Meta.GetName()
+	klog.V(10).Info("In cluster Mapper for ", cname)
+
+	var requests []reconcile.Request
+
+	dplList := &appv1alpha1.DeployableList{}
+
+	listopts := &client.ListOptions{}
+	err := mapper.List(context.TODO(), listopts, dplList)
+	if err != nil {
+		klog.Error("Failed to list deployables for cluster mapper with error:", err)
+		return requests
+	}
+
+	for _, dpl := range dplList.Items {
+		// only reconcile with when placement is set and not using ref
+		if dpl.Spec.Placement == nil || dpl.Spec.Placement.PlacementRef == nil {
+			continue
+		}
+
+		if dpl.Spec.Placement.ClusterSelector == nil {
+			matched := false
+			for _, cn := range dpl.Spec.Placement.Clusters {
+				if cn.Name == cname {
+					matched = true
+				}
+			}
+			if !matched {
+				continue
+			}
+		}
+		// only reconcile it's own object
+		annotations := dpl.GetAnnotations()
+		if annotations == nil || annotations[appv1alpha1.AnnotationManagedCluster] == "" {
+			continue
+		}
+
+		if annotations[appv1alpha1.AnnotationManagedCluster] != (types.NamespacedName{}).String() {
```

Comment:
Same as line 101, Should it be 
```
if annotations[appv1alpha1.AnnotationManagedCluster] == (types.NamespacedName{}).String()  {
    continue
}
```

Reply:
this one is correct. != "\" means it is a deployable for managed cluster. we shall skip (continue)

Reply:
see the reply in line 101. This is correct. (types.NamespacedName{}).String() = '\'